File: app/src/main/java/com/hiteshsahu/awesome_gallery/view/adapter/MediaViewHolder.java
package com.hiteshsahu.awesome_gallery.view.adapter;

import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.hiteshsahu.awesome_gallery.R;


public class MediaViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    private TextView mediaName, mediaDesc;
    private ImageView mediaImage;
    private CardView container;
    private LinearLayout containerbg;
    private View menu;


    public LinearLayout getContainerbg() {
        return containerbg;
    }

    /**
     * @param v
     */
    public MediaViewHolder(View v) {
        super(v);
        mediaName = (TextView) v.findViewById(R.id.mediaName);
        mediaDesc = (TextView) v.findViewById(R.id.product_desc);
        mediaImage = (ImageView) v.findViewById(R.id.product_image);
        container = (CardView) v.findViewById(R.id.list_container);
        containerbg = (LinearLayout) v.findViewById(R.id.list_container_bg);
        menu = (View) v.findViewById(R.id.overflow);
    }

    public ImageView getMediaImage() {
        return mediaImage;
    }

    public TextView getMediaName() {
        return mediaName;
    }

    public TextView getMediaDesc() {
        return mediaDesc;
    }

    public CardView getContainer() {
        return container;
    }

    public void clearAnimation() {
        container.clearAnimation();
    }

    @Override
    public void onClick(View view) {

    }

    public View getMenu() {
        return menu;
    }
}

File: app/src/main/java/com/hiteshsahu/awesome_gallery/view/adapter/MediaViewHolder.kt
package com.hiteshsahu.awesome_gallery.view.adapter

import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.hiteshsahu.awesome_gallery.R


class MediaViewHolder
/**
 * @param v
 */
(v: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(v), View.OnClickListener {

    val mediaName: TextView = v.findViewById<View>(R.id.mediaName) as TextView
    val mediaDesc: TextView = v.findViewById<View>(R.id.media_location) as TextView
    val mediaImage: ImageView = v.findViewById<View>(R.id.image) as ImageView
    val container: androidx.cardview.widget.CardView = v.findViewById<View>(R.id.list_container) as androidx.cardview.widget.CardView
    val containerbg: LinearLayout = v.findViewById<View>(R.id.list_container_bg) as LinearLayout
    val menu: View = v.findViewById(R.id.overflow) as View

    fun clearAnimation() {
        container.clearAnimation()
    }

    override fun onClick(view: View) {

    }
}